client/javascript-esm/packages/jolokia/src/jolokia-utils.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  JolokiaErrorResponse,
+  JolokiaSuccessResponse,
+  VersionResponseValue
+} from "./jolokia-types.js"
+
+export type ParseResult<T> = { hasError: false; parsed: T } | { hasError: true; error: string }
+
+function isObject(value: unknown): value is object {
+  const type = typeof value
+  return value != null && (type === 'object' || type === 'function')
+}
+
+export function isResponseSuccessType(o: unknown): o is JolokiaSuccessResponse {
```

Comment:
The function name seems to be a bit redundant. Considering we already have `isError(resp: JolokiaResponse)`, maybe `isResponseSuccess(o: unknown)` is better.